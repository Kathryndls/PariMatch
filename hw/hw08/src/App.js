import React from "react";
import Scoreboard from "./components/Scoreboard/Scoreboard";
import SearchInput from "./components/SearchInput/SearchInput";
import WeeklyForecast from "./components/WeeklyForecast/WeeklyForecast";

import Container from './hoc/Container/Container';

import classes from './App.module.scss';
import './index.module.scss';

function App() {
	return (
		<Container>
			<h1 className={classes['Title']}>Weather Forecast by Katerina Kolesnykova</h1>

			<div className={classes['RowHeader']}>
				<SearchInput />
				<Scoreboard />
			</div>
			<WeeklyForecast />
		</Container>
	);
}

export default App;