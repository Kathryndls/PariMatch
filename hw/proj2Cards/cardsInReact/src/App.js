import { useState, useEffect } from "react";
import {Routes} from './routes';
import AuthService from "./services/AuthService";
import Login from "./components/Login";
import Register from "./components/Register";
import Test from "./components/Test";
import { Header } from "./components/Header";
import Dashboard from "./components/Dashboard";
import { Modal } from "./components/Modal";
import getStatuses from "./services/get-statuses";
// // import {Header} from "./components/Header/Header";
// import Dashboard from "./components/Dashboard/Dashboard";

function App() {
  const [currentUser, setCurrentUser] = useState(undefined);
  const [isVisible, setIsVisible] = useState(false);
  
  useEffect(() => {
    const user = AuthService.getCurrentUser();

    if (user) {
      setCurrentUser(user);
    }

    getStatuses.get()
  }, []);

  const logOut = () => {
    AuthService.logout();
  };
  
  return (
    <>
      <Header currentUser={currentUser} setCurrentUser={setCurrentUser} />
      <Routes currentUser={currentUser} />

      <Modal isVisible={isVisible} onHide={setIsVisible} />
    </>
  );
}

export default App;
