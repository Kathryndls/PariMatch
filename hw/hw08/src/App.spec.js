import React from "react";
import App from "./App";

const setUp = (props) => shallow(<App {...props} />);

describe('should render App component', () => {
   let component;
   beforeEach(() => {
      component = setUp();
   })

   it ('should have a Title class', () => {
      const wrapper = component.find('.Title');
      expect(wrapper.length).toBe(1);
   });

   it ('should have a RowHeader class', () => {
      const wrapper = component.find('.RowHeader');
      expect(wrapper.length).toBe(1);
   })

   it ('should have a Container component', () => {
      const wrapper = component.find('Container');
      expect(wrapper.length).toBe(1);
   });

   it ('should have a Scoreboard component', () => {
      const wrapper = component.find('Scoreboard');
      expect(wrapper.length).toBe(1);
   });

   it ('should have a SearchInput component', () => {
      const wrapper = component.find('SearchInput');
      expect(wrapper.length).toBe(1);
   });

   it ('should have a WeeklyForecast component', () => {
      const wrapper = component.find('WeeklyForecast');
      expect(wrapper.length).toBe(1);
   });


   it ('should have a h1 tag', () => {
      const wrapper = component.find('h1');
      expect(wrapper.length).toBe(1);
   })

   it ('should have a div tag', () => {
      const wrapper = component.find('div');
      expect(wrapper.length).toBe(1);
   })
});

describe('App component', () => {
   it ('should render App component', () => {
      const component = setUp();
      expect(component).toMatchSnapshot();
   })
})