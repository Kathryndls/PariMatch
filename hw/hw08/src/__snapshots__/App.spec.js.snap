// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `
<Container>
  <h1
    className="Title"
  >
    Weather Forecast by Katerina Kolesnykova
  </h1>
  <div
    className="RowHeader"
  >
    <SearchInput />
    <Scoreboard />
  </div>
  <WeeklyForecast />
</Container>
`;
